docs/new-project-setup.md

```diff
@@ -0,0 +1,315 @@
+    * The service account name
+    * The public key
+    * The private key
+
+1. Create an IT Help Desk ticket to add the new account to our Sentinel instance.
+    Share the 1Password item with the IT-Ops staff member who is implementing the ticket.
+    If you've included all of the above information in the vault item,
+    it should be all they need.
+1. Within fifteen minutes or so of implementation it should be clear whether the integration is working.
+    IT-Ops should be able to see logs ingesting, and Snowflake account admins should see queries
+    from the Sentinel service user.
+
+## Set up CI in GitHub
+
+The projects generated from our infrastructure template need read access to the
+Snowflake account in order to do two things from GitHub actions:
+
+1. Verify that dbt models in branches compile and pass linter checks
+1. Generate dbt docs upon merge to `main`.
+
+The terraform configurations deployed above create two service accounts
+for GitHub actions, a production one for docs and a dev one for CI checks.
+
+### Add key pairs to the GitHub service accounts
+
+Set up key pairs for the two GitHub actions service accounts
+(`GITHUB_ACTIONS_SVC_USER_DEV` and `GITHUB_ACTIONS_SVC_USER_PRD`).
+This follows a similar procedure to what you did for your personal key pair,
+though the project template currently does not assume an encrypted key pair.
+[This bash script](https://gist.github.com/ian-r-rose/35d49bd253194f57b57e9e59a595bed8)
+is a helpful shortcut for generating the key pair:
+```bash
+bash generate_key.sh <key-name>
+```
+
+Once you have created and set the key pairs, add them to the DSE 1Password shared vault.
+Make sure to provide enough information to disambiguate the key pair from others stored in the vault,
+including:
+
+* The account locator
+* The account name (distinct from the account locator)
+* The service account name
+* The public key
+* The private key
+
+### Set up GitHub actions secrets
+
+You need to configure secrets in GitHub actions
+in order for the service accounts to be able to connect to your Snowflake account.
+From the repository page, go to "Settings", then to "Secrets and variables", then to "Actions".
+
+Add the following repository secrets:
+
+| Variable | Value |
+|----------|-------|
+| `SNOWFLAKE_ACCOUNT` | new account locator |
+| `SNOWFLAKE_USER_DEV` | `GITHUB_ACTIONS_SVC_USER_DEV` |
+| `SNOWFLAKE_USER_PRD` | `GITHUB_ACTIONS_SVC_USER_PRD` |
+| `SNOWFLAKE_PRIVATE_KEY_DEV` | dev service account private key |
+| `SNOWFLAKE_PRIVATE_KEY_PRD` | prd service account private key |
+
+### Enable GitHub pages for the repository
+
+The repository must have GitHub pages enabled in order for it to deploy and be viewable.
+
+1. From the repository page, go to "Settings", then to "Pages".
+1. Under "GitHub Pages visibility" select "Private" (unless the project is public!).
+1. Under "Build and deployment" select "Deploy from a branch" and choose "gh-pages" as your branch.
```

Comment:
should there be a "gh-pages" branch already? bc there isn't one for me @ian-r-rose

Reply:
It should be there after the first time the docs action successfully completes

Reply:
My GH actions fail due to an inability to install logbook. >> "Cannot install logbook." Is that expected?

Reply:
do you have a link?

Reply:
https://github.com/cagov/britt_test/actions/runs/6775785678/job/18418081852#step:7:122 @ian-r-rose

Reply:
added you to the repo @ian-r-rose

Reply:
Try committing your `poetry.lock` file to the repo

Reply:
I don't have that file. I am not seeing it mentioned in the set up docs either.

Reply:
We can add it here, there's some discussion of poetry in the local setup docs but not here.

I'm also not certain it will fix the issue you are seeing. Try running `poetry install`, that should generate the file

Reply:
new errors: https://github.com/cagov/britt_test/actions/runs/6776890533/job/18419372049#step:8:30
and: https://github.com/cagov/britt_test/actions/runs/6776890551 @ian-r-rose

Reply:
Looks like the credentials in the repo secrets are not right (that's dbt complaining that it can't connect to Snowflake)

Reply:
are we the value for `SNOWFLAKE_USER_DEV` to literally be `GITHUB_ACTIONS_SVC_USER_DEV` and the same for prd? @ian-r-rose